src/app/core/services/ledger.service.ts

```diff
@@ -1,30 +1,35 @@
 import { Injectable } from '@angular/core';
-import { runTransaction } from '@numbersprotocol/niota';
-import { from, Observable, of } from 'rxjs';
-import { catchError, tap, timeout } from 'rxjs/operators';
+import { sendMessage } from '@numbersprotocol/niota';
+import { Observable, of } from 'rxjs';
+import { catchError, map, switchMap, take, tap, timeout } from 'rxjs/operators';
+import { WebCryptoApiSignatureProvider } from '../../core/services/web-crypto-api-signature-rovider.service'
 
 @Injectable({
   providedIn: 'root',
 })
 export class LedgerService {
-  constructor() {}
+  constructor(
+    private readonly CryptoSignature: WebCryptoApiSignatureProvider
+  ) {
+    this.CryptoSignature.initialize();
+  }
 
   private registerOnLedger(hash: string): Observable<any> {
-    const address =
-      'HEQLOWORLDHELLOWORLDHELLOWORLDHELLOWORLDHELLOWORLDHELLOWORLDHELLOWORLDHELLOWOR99D';
-    const seed =
-      'PUEOTSEITFEVEWCWBTSIZM9NKRGJEIMXTULBACGFRQK9IMGICLBKW9TTEVSDQMGWKBXPVCBMMCXWMNPDX';
+    const index = 'Lifebox';
     const rawMsg = { hash };
-    return from(runTransaction(address, seed, rawMsg)).pipe(
+    return this.CryptoSignature.getCryptoData().pipe(
+      switchMap((dummySignature) => sendMessage(index, rawMsg, dummySignature.publicKey)),
```

Comment:
```suggestion
      concatMap((cryptoData) => sendMessage(index, rawMsg, cryptoData.publicKey)),
```

It's not a dummy signature, use a better name that fits the actual data.
Also concatMap is preferred over switchMap when there's no need for flattening the inner observable emission.